src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/wrapper/ErrorHandlingWrapper.java

```diff
@@ -50,4 +53,30 @@ public void wrapDocUploadFailure(Envelope envelope, Supplier<Void> supplier) {
             errorHandler.handleError(exc);
         }
     }
+
+    public <T> T wrapAcquireLeaseFailure(
+        String containerName,
+        String zipFileName,
+        Supplier<T> supplier
+    ) {
+        try {
+            return supplier.get();
+        } catch (Exception exception) {
+
+            boolean isLeaseAlreadyAcquired = Optional.of(exception)
+                .filter(e -> e instanceof StorageException)
+                .map(se -> (StorageException) se)
+                .filter(se -> se.getHttpStatusCode() == HttpStatus.CONFLICT.value())
+                .isPresent();
+
+            Exception exc = isLeaseAlreadyAcquired
+                ? new LeaseAlreadyPresentException(
+                "Lease already acquired for container " + containerName + " and zip file " + zipFileName)
+                : exception;
```

Comment:
perhaps this would be more idiomatic
```
if (exception instanceof StorageException && ((StorageException) exception).getHttpStatusCode == CONFLICT) {
    errorHandler.handleError(new LeaseAlreadyPresentException(...));
} else {
    errorHandler.handleError(exc);
}

Reply:
there's no benefit of try/catch signature. i would still vote for ^ approach but let's consider this one too:

```java
try {
  // smth to return
} catch (StorageException exception) {
    // more functionally comparison but it doesn't matter
    boolean isLeaseAcquired = HttpStatus.valueOf(exception.getCode()).equals(CONFLICT));

    errorHandler.handleError(isLeaseAcquired ? new LeaseAlreadyPresentException(...) : exception);
} catch (Exception exception) {
    errorHandler.handleError(exception);
}

return null;
```

p.s. the event will not be registered via error handler since exceptions are not `EventAwareThrowable`s. I think it is intended so all good. Just a note here